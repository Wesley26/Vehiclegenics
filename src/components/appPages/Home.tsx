import { FC } from 'react';
import { globalStyles } from '../../../globalStyles';
import { useTailwind } from 'tailwind-rn';
import { LinearGradient } from 'expo-linear-gradient';

import { 
    MainLogo,
} from '../globalComponents/indexGlobal';

const Home:FC = () => {

    const tailwind = useTailwind();

    return (
        <LinearGradient
            colors={[
                globalStyles.primary.color, 
                globalStyles.textMedium.color, 
                globalStyles.textDark.color
            ]}  
            style={tailwind('p-20pix flex-1 flex-col')}
        >

            <MainLogo />

        </LinearGradient>
    );
};

export default Home;