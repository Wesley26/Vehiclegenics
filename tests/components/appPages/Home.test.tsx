import React from "react";
import renderer from "react-test-renderer";
import Home from '../../../src/components/appPages/Home';

//required so tailwindcss + tailwind-rn play nice with test
jest.useFakeTimers();

describe (`describing Home.tsx component`, () => {

    it(`render Home.tsx test component`, () => {

        const view = renderer.create(<Home />);
        expect(view).toMatchSnapshot();

    });

});