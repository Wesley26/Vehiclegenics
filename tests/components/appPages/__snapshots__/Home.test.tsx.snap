// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describing Home.tsx component render Home.tsx test component 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4288520633,
        4284835702,
        4282598478,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={Object {}}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#656776",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "color": "#43444e",
          },
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "WorkSans_700Bold",
            },
            Object {
              "textShadowOffset": Object {
                "height": 2,
                "width": -2,
              },
              "textShadowRadius": 1,
            },
            Object {},
          ]
        }
      >
        Vehiclegenics
      </Text>
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
